airbyte-cdk/python/airbyte_cdk/sources/streams/call_rate.py

```diff
@@ -0,0 +1,445 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import abc
+import dataclasses
+import datetime
+import logging
+import time
+from datetime import timedelta
+from typing import TYPE_CHECKING, Any, Mapping, Optional
+from urllib import parse
+
+import requests
+import requests_cache
+from pyrate_limiter import InMemoryBucket, Limiter
+from pyrate_limiter import Rate as PyRateRate
+from pyrate_limiter import RateItem, TimeClock
+from pyrate_limiter.exceptions import BucketFullException
+
+if TYPE_CHECKING:
```

Comment:
can you add a comment explaining why we need this if/else? I think I've seen this for the requests_caching mixin too

Reply:
this is for mypy again, to prevent it complaining about missing Session attributes